ls1intum/Artemis
# `Programming exercises`: Fix "Go to Build Plan" URL in the exercise overview

### Checklist
#### General
#### Client

### Motivation and Context
Closes #5431

### Description
The buildPlanURL was undfined in the reported case. It's a helper attribute and needs to be calculated in the client. I added a call to `setBuildPlanUrlForProgrammingParticipations` to calculate the needed value. 

### Steps for Testing
Replicate the issue. 

Prerequisites:
- 1 Programming Exercise with "Publish Build Plans" enabled

1. Navigate to the exercise overview
2. Start the exercise -> the button should appear and the "Go to Build Plan" link should work
3. Click on the exercise detail -> link should work
4. As instructor, cleanup the participation and resume as student -> link should work
5. When the exercise is shown as a "featured exercise" on the front page, the link should work

### Review Progress





#### Code Review
#### Manual Tests
